its/src/test/java/com/sonar/it/csharp/MetricsIncludeHeaderCommentTest.java

```diff
@@ -0,0 +1,201 @@
+ * Copyright (C) 2011-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.csharp;
+
+import com.sonar.it.shared.TestUtils;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.SystemUtils;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.sonarqube.ws.WsMeasures.Measure;
+
+import static com.sonar.it.csharp.Tests.ORCHESTRATOR;
+import static com.sonar.it.csharp.Tests.getComponent;
+import static com.sonar.it.csharp.Tests.getMeasure;
+import static com.sonar.it.csharp.Tests.getMeasureAsInt;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This is copy pasted from MetricsTest. It does not re-test everything, it just makes sure
+ * that when 'ignoreHeaderComments' is set to False, it counts the header comments as well
+ * and it does not modify the LOC metrics.
+ */
+public class MetricsIncludeHeaderCommentTest {
+
+  public static TemporaryFolder temp = new TemporaryFolder();
+
+  private static final String PROJECT = "MetricsTest";
+  private static final String DIRECTORY = TestUtils.hasModules(ORCHESTRATOR) ? "MetricsTest:MetricsTest:1F026ECA-900A-488D-9D07-AD23216FA32B:foo" : "MetricsTest:foo";
+  private static final String FILE = TestUtils.hasModules(ORCHESTRATOR) ? "MetricsTest:MetricsTest:1F026ECA-900A-488D-9D07-AD23216FA32B:foo/Class1.cs" : "MetricsTest:foo/Class1.cs";
+  private static final int NUMBER_OF_HEADER_COMMENT_LINES = 2;
+
+  @ClassRule
+  public static RuleChain chain = getRuleChain();
+
+  private static RuleChain getRuleChain() {
+    assertThat(SystemUtils.IS_OS_WINDOWS).withFailMessage("OS should be Windows.").isTrue();
+
+    // Scanner for MSBuild caches the analyzer, so running the test twice in a row means the old binary is used.
+    // This code deletes the cache, but there should be a way to run without cache.
+    // Ticket: https://jira.sonarsource.com/browse/SONARMSBRU-346
```

Comment:
this has actually been closed as "Won't fix" https://jira.sonarsource.com/browse/SONARMSBRU-346